server/audit_math/supersimple.py

```diff
@@ -40,6 +51,82 @@ def nMin(
     )
 
 
+def compute_discrepancies(
+    contest: Contest, cvrs: CVRS, sample_cvr: CVRS,
+) -> Dict[str, Discrepancy]:
+    """
+    Iterates through a given sample and returns the discrepancies found.
+
+    Inputs:
+        contests       - the contests and results being audited
+        cvrs           - mapping of ballot_id to votes:
+                {
+                    'ballot_id': {
+                        'contest': {
+                            'candidate1': 1,
+                            'candidate2': 0,
+                            ...
+                        }
+                    ...
+                }
+
+        sample_cvr - the CVR of the audited ballots
+                {
+                    'ballot_id': {
+                        'contest': {
+                            'candidate1': 1,
+                            'candidate2': 0,
+                            ...
+                        }
+                    ...
+                }
+
+    Outputs:
+        discrepancies   - A mapping of ballot ids to their discrepancies. This
+                          includes entries for all ballots in the sample, even those without
+                          discrepancies.
+                {
+                    'ballot_id': {
+                        'counted_as': -1, # The maximum value for a discrepancy
+                        'weighted_error': -0.0033 # Weighted error used for p-value calculation
+                        'cvr': <CVR for this ballot>,
+                        'sample_cvr': <Sample CVR for this ballot>
+                    }
+    """
+
+    discrepancies: Dict[str, Discrepancy] = {}
+    for ballot in sample_cvr:
+        e_r = 0.0
+        e_int = 0
```

Comment:
my understanding is that `e_r` and `e_int` are used to compute maximum error values for the ballot across the different winner/loser combos. if that's the case, should their initial values be the minimum possible error value (which I believe is `-2` for `e_int`, not sure what it would be for `e_r`)?

Reply:
In order to remain as conservative as possible, we're choosing to ignore negative errors, so that they don't cancel out positive errors.

Reply:
Oh that's super interesting. Can you add a comment explaining that?

Also, does that mean that in the audit report we'll end up showing a discrepancy of 0 for ballots that upon examination would appear to have a negative discrepancy? Will that be confusing for people? Or is that expected behavior?

Reply:
Will add a comment. @mcchilders what do you think about this?

Reply:
We "ignore" them only for the purposes of calculating the p-value (we still record them all as discrepancies, and list them out that way in the audit report, etc. Each discrepancy has its own row - it's not a combined discrepancy count per ballot, where the combination might hide individual errors, it's literally a list of each discrepancy and which ballot it appeared on.

Reply:
Hmm, that means either I need to be doing more work in the math code, or @jonahkagan will need to check for the discrepancies when writing the audit report. Which of those sounds better to you, Jonah?